tutorials/RandomNumberGeneration/Tests.qs

```diff
@@ -17,65 +17,108 @@ namespace Quantum.Kata.RandomNumberGeneration {
     open Microsoft.Quantum.Random;
 
     // Exercise 1.
-    @Test("QuantumSimulator")
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T1_RandomBit () : Unit {
-        Message("Testing...");
-        CheckFlatDistribution(RandomBit_Wrapper, 1, 0.4, 0.6, 1000, 450);
+        Message("Testing One RandomBitGeneration...");
+        RetryCheckUniformDistribution(RandomBit_Wrapper, 0, 1, 1000, 3);
+        Message("One random bit successfully generated...");
     }
 
-    operation RandomBit_Wrapper (throwaway: Int) : Int {
+    operation RandomBit_Wrapper (throwawayMin: Int, throwawayMax: Int) : Int {
         return RandomBit();
     }
 
 
     // Exercise 2.
-    @Test("QuantumSimulator")
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T2_RandomTwoBits () : Unit {
-        Message("Testing...");
-        CheckFlatDistribution(RandomTwoBits_Wrapper, 2, 1.4, 1.6, 1000, 200);
+        Message("Testing Two RandomBitGeneration...");
+        RetryCheckUniformDistribution(RandomTwoBits_Wrapper, 0, 3, 1000, 3);
+        Message("Two random bits successfully generated...");
     }
 
-    operation RandomTwoBits_Wrapper (throwaway: Int) : Int {
+    operation RandomTwoBits_Wrapper (throwawayMin: Int, throwawayMax: Int) : Int {
         return RandomTwoBits();
     }
     
 
     // Exercise 3.
-    @Test("QuantumSimulator")
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T3_RandomNBits () : Unit {
         Message("Testing N = 1...");
-        CheckFlatDistribution(RandomNBits, 1, 0.4, 0.6, 1000, 450);
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 1, 1000, 3);
```

Comment:
In this case the same sequence of 3 commands repeats with only two parameters varying. I would recommend to replace copy-paste with a loop: `for (min, max) in [(0, 1), (0, 3), (0, 7), (0, 1023)] { ... }`, and use `min` and `max` as parameters for those three commands inside the loop body. This will make the code much easier to maintain, for example, if we decide to change one of the parameter sets, we'll only need to do it in one place rather than in 3.

The same suggestion goes for repeated tests in `T5_RandomNumberInRange`.

Reply:
Done